narwhals/expr.py

```diff
@@ -1833,6 +1833,50 @@ class ExprStringNamespace:
     def __init__(self, expr: Expr) -> None:
         self._expr = expr
 
+    def strip_chars(self, characters: str | None = None) -> Expr:
+        r"""
+        Remove leading and trailing characters.
+
+        Arguments:
+            characters: The set of characters to be removed. All combinations of this set of characters will be stripped from the start and end of the string. If set to None (default), all leading and trailing whitespace is removed instead.
+
+        Examples:
+            >>> import pandas as pd
+            >>> import polars as pl
+            >>> import narwhals as nw
+            >>> data = {"fruits": ["apple", "\nmango"]}
+            >>> df_pd = pd.DataFrame(data)
+            >>> df_pl = pl.DataFrame(data)
+
+            We define a dataframe-agnostic function:
+
+            >>> @nw.narwhalify
+            ... def func(df):
+            ...     return df.with_columns(stripped=nw.col("fruits").str.strip_chars())
+
+            We can then pass either pandas or Polars to `func`:
+
+            >>> func(df_pd)
+                fruits stripped
+            0    apple    apple
+            1  \nmango    mango
+
+            >>> func(df_pl)
+            shape: (2, 2)
+            ┌────────┬──────────┐
+            │ fruits ┆ stripped │
+            │ ---    ┆ ---      │
+            │ str    ┆ str      │
+            ╞════════╪══════════╡
+            │ apple  ┆ apple    │
+            │        ┆ mango    │
+            │ mango  ┆          │
```

Comment:
This rendering hit me so hard 🙈
Shall we use another "whitespace" character instead of a new line?

Reply:
ikr 😂, I was copying from polar's [Expr.str.strip_chars](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.str.strip_chars.html#polars.Expr.str.strip_chars) documentation.. I think it's because "\n" is the most _obvious_ trailing whitespace.

See for example how "\t" is not that obvious in the repr 
```pycon
>>> df = pl.DataFrame({"col_a": "mango\t", "col_b":"mango"})
>>> df
shape: (1, 2)
┌────────┬───────┐
│ col_a  ┆ col_b │
│ ---    ┆ ---   │
│ str    ┆ str   │
╞════════╪═══════╡
│ mango	 ┆ mango │
└────────┴───────┘
```
Maybe there is a better way to represent it ? Possibly with a final `to_dict` ?

Reply:
yeah `to_dict(as_series=False)` might be good! and maybe even a good idea for an upstream PR to Polars if you fancy it 😉